extensions/standard-processors/tests/unit/SegmentContentTests.cpp

```diff
@@ -0,0 +1,267 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <random>
+
+#include "FlowFileRecord.h"
+#include "catch2/generators/catch_generators.hpp"
+#include "processors/SegmentContent.h"
+#include "range/v3/algorithm/equal.hpp"
+#include "unit/Catch.h"
+#include "unit/SingleProcessorTestController.h"
+#include "unit/TestBase.h"
+#include "range/v3/algorithm/generate.hpp"
+
+namespace org::apache::nifi::minifi::processors::test {
+
+std::vector<std::byte> generateRandomData(const size_t n) {
+  std::independent_bits_engine<std::default_random_engine, CHAR_BIT, uint16_t> rbe{gsl::narrow_cast<uint16_t>(std::chrono::system_clock::now().time_since_epoch().count())};
+  std::vector<std::byte> bytes(n);
+  ranges::generate(bytes, [&]() { return static_cast<std::byte>(rbe()); });
+  return bytes;
+}
+
+std::string_view calcExpectedSegment(const std::string_view original_content, const size_t segment_i, const size_t segment_size) {
+  const auto start_pos = segment_i * segment_size;
+  const auto end_pos = std::min(start_pos + segment_size, original_content.length());
+  const auto actual_size = std::min(segment_size, end_pos - start_pos);
+  return original_content.substr(segment_i * segment_size, std::min(segment_size, actual_size));
+}
+
+std::span<const std::byte> calcExpectedSegment(const std::span<const std::byte> original_content, const size_t segment_i, const size_t segment_size) {
+  const auto start_pos = segment_i * segment_size;
+  const auto end_pos = std::min(start_pos + segment_size, original_content.size());
+  const auto actual_size = std::min(segment_size, end_pos - start_pos);
+  return original_content.subspan(segment_i * segment_size, std::min(segment_size, actual_size));
+}
+
+template<typename... Bytes>
+std::vector<std::byte> createByteVector(Bytes... bytes) {
+  return {static_cast<std::byte>(bytes)...};
+}
+
+TEST_CASE("Invalid segmentSize tests") {
+  const auto segment_content = std::make_shared<SegmentContent>("SegmentContent");
+  minifi::test::SingleProcessorTestController controller{segment_content};
+
+  SECTION("foo") {
+    REQUIRE_NOTHROW(segment_content->setProperty(SegmentContent::SegmentSize, "foo"), "General Operation: Segment Size value validation failed");
+    REQUIRE_THROWS_WITH(controller.trigger("bar"), "Processor Operation: Invalid Segment Size: 'foo'");
+  }
+  SECTION("-1") {
+    REQUIRE_NOTHROW(segment_content->setProperty(SegmentContent::SegmentSize, "-1"), "General Operation: Segment Size value validation failed");
+    REQUIRE_THROWS_WITH(controller.trigger("bar"), "narrowing_error");
+  }
```

Comment:
narrowing errors should be treated as `terminate` calls, not exceptions. So I would remove this test case.

Reply:
👍 [review changes](https://github.com/apache/nifi-minifi-cpp/pull/1879/commits/e830b8b7af003b3ffb0d19b1620f713bd3fb6d28)